packages/postgres/src/PostgresMessagePickupRepository.ts

```diff
@@ -0,0 +1,569 @@
+import {
+  AddMessageOptions,
+  Agent,
+  GetAvailableMessageCountOptions,
+  injectable,
+  Logger,
+  MessagePickupEventTypes,
+  MessagePickupLiveSessionRemovedEvent,
+  MessagePickupLiveSessionSavedEvent,
+  MessagePickupRepository,
+  QueuedMessage,
+  RemoveMessagesOptions,
+  TakeFromQueueOptions,
+} from '@credo-ts/core'
+import { Pool, Client } from 'pg'
+import PGPubsub from 'pg-pubsub'
+import * as os from 'os'
+import { createTableMessage, createTableLive, messagesTableName, liveSessionTableName } from '../config/dbCollections'
+import { ConnectionInfo, PostgresMessagePickupRepositoryConfig } from './interfaces'
+import { MessagePickupSession } from '@credo-ts/core/build/modules/message-pickup/MessagePickupSession'
+import axios from 'axios'
+
+@injectable()
+export class PostgresMessagePickupRepository implements MessagePickupRepository {
+  private logger?: Logger
+  public messagesCollection?: Pool
+  private agent?: Agent
+  private pubSubInstance?: PGPubsub
+  private dbListener?: boolean
+  private instanceName?: string
+  private postgresUser: string
+  private postgresPassword: string
+  private postgressHost: string
+  private postgresDatabaseName: string
+  private connectionInfoCallback?: (connectionId: string) => Promise<ConnectionInfo | undefined>
+
+  public constructor(options: PostgresMessagePickupRepositoryConfig) {
+    const { logger, postgresUser, postgresPassword, postgressHost, postgresNameDatabase, fcmServiceBaseUrl } = options
+
+    this.logger = logger
+    this.postgresUser = postgresUser
+    this.postgresPassword = postgresPassword
+    this.postgressHost = postgressHost
+    this.postgresDatabaseName = postgresNameDatabase || 'messagepickuprepository'
+  }
+
+  /**
+   * Initializes the service by setting up the database, message listeners, and the agent.
+   * This method also configures the Pub/Sub system and registers event handlers.
+   *
+   * @param {Agent} agent - The agent instance to be initialized.
+   * @param {(connectionId: string) => Promise<ConnectionInfo | undefined>} connectionInfoCallback -
+   * A callback function that retrieves connection information for a given connection ID.
+   * @returns {Promise<void>} A promise that resolves when the initialization is complete.
+   * @throws {Error} Throws an error if initialization fails due to database, Pub/Sub, or agent setup issues.
+   */
+  public async initialize(
+    agent: Agent,
+    connectionInfoCallback: (connectionId: string) => Promise<ConnectionInfo | undefined>,
```

Comment:
This can be also encapsulated in an options object:

```suggestion
  public async initialize(options: {
    agent: Agent,
    connectionInfoCallback: (connectionId: string) => Promise<ConnectionInfo | undefined>}
```

Reply:
solved!